### ISSUE TITLE
Go backend only allows setting up one roll call at a time

### ISSUE THREAD

**Issue by Kar1213** [dedis/student_21_pop#353]
The problem is that we can only use the roll call that has been created last. I noticed the problem when sending requests from fe2, for example if after creating two roll calls we want to open the first one, the server refuses saying that the roll call needs to be created before opening it. This problem occurs similarly when we want to reopen a roll call after closing another. My guess is that if the server receives a roll call action other than a creation, it checks that the updateId of the action corresponds to the updateId of the last roll call action it received. 
Is the behavior that is actually expected or is this indeed a bug as it sounds a bit restrictive?

**Comment by flmichel** [comment]
Yes I agree this is restrictive! I'm going to do the necessary modifications do be able to create multiple roll call in parallel.

**Comment by flmichel** [comment]
After thinking about it, I'm not sure it is a good idea to allow multiple roll-calls even if we check that the roll-calls do not overlap and if we prevent to open multiple roll-calls at the same time. An organizer might still be able to run 2 roll-calls at the same time if he keep closing/reopening each roll-call one after the other. He would be able to a scan some participants public key in a roll call and some other public key in the other roll-call. And, I think this is a problem.

**Comment by Kar1213** [comment]
For closing/reopening different roll-calls I agree that this might not be something we would want, but the problem is more general, for example we can't create 2 rollcalls and then open the first one (we can only open the last one).

**Comment by pierluca** [comment]
There are multiple problems raised by the discussion here.
1. From a PoP perspective, simultaneous open roll calls do not make sense, for all the reasons @flmichel listed.
2. It should thus be forbidden to have multiple roll calls whose proposed start/end dates overlap (to be specified in the protocol)
3. It should similarly be forbidden ot have multiple roll calls open at the same time. (to be specified in the protocol)
4. However, pre-planned roll calls should be fully supported by all sub-systems.

**Comment by lourkeur** [comment]
Confirmed to still occur on db6b5c0a9, on both back-ends.

1. create two roll calls
2. cannot open the first created one.

BE1: `invalid message field: failed to publish: failed to handle publish message: failed to process message: failed to process action 'roll_call#open': The field `opens` does not correspond to the id of the previous roll call message`
BE2: `OpenRollCall content validation failed: unexpected id 'opens'`
